src/common/transformations/src/transformations/common_optimizations/optimize_strided_slice.cpp

```diff
@@ -274,30 +274,47 @@ struct SliceWithAttrs {
 };
 
 bool slice_is_suitable_for_optimization(const std::shared_ptr<ov::op::v8::Slice>& op, SliceAttrs& attrs) {
-    const auto& data_rank = op->get_input_partial_shape(0).rank();
+    const auto& input_shape = op->get_input_partial_shape(0);
+    const auto& data_rank = input_shape.rank();
     if (op->get_input_size() != 5 || data_rank.is_dynamic())
         return false;
+    const auto rank = data_rank.get_length();
 
-    for (size_t i = 1; i < 5; ++i) {
-        auto input_as_constant = ov::as_type_ptr<ov::op::v0::Constant>(op->get_input_node_shared_ptr(i));
-        if (!input_as_constant)
+    auto get_scalar = [](const std::shared_ptr<ov::Node>& node, int64_t& value) -> bool {
+        auto constant = ov::as_type_ptr<ov::op::v0::Constant>(node);
+        if (!constant)
             return false;
-        if (shape_size(input_as_constant->get_shape()) != 1)
+        if (shape_size(constant->get_shape()) != 1)
             return false;
+        value = constant->cast_vector<int64_t>()[0];
+        return true;
+    };
 
-        int64_t value = input_as_constant->cast_vector<int64_t>()[0];
+    if (!get_scalar(op->get_input_node_shared_ptr(4), attrs.axis))
+        return false;
+    attrs.axis = attrs.axis >= 0 ? attrs.axis : attrs.axis + rank;
 
-        if (((i == 1 || i == 2) && value < 0) || (i == 3 && value != 1))
+    if (input_shape[attrs.axis].is_dynamic())
+        return false;
+    const auto dimension = input_shape[attrs.axis].get_length();
+
+    for (int i = 1; i < 4; i++) {
+        int64_t value;
+        if (!get_scalar(op->get_input_node_shared_ptr(i), value))
             return false;
-        else if (i == 1)
+        if (i == 3) {
+            if (value != 1)
+                return false;
+            continue;
+        }
+        value = value >= 0 ? value : value + dimension;
```

Comment:
I think the previous implementation had better readability.
could we use "get_scalar(" lambda to process negative values?
I think we need to pass "rank" or "dimension" to this lambda as an additional argument
maybe somethink like that
vector<size_t> right_border_values = {dim, dim, dim, dim, rank};
after I hope we will be able to handle all inputs in one loop.
Could you also use unnamed enum instead of indexes for inputs?
enum { BEGIN = 1, END, STRIDE, AXIS };
if (BEGIN == i) etc

Reply:
> could we use "get_scalar(" lambda to process negative values?
> I think we need to pass "rank" or "dimension" to this lambda as an additional argument
> maybe somethink like that
> vector<size_t> right_border_values = {dim, dim, dim, dim, rank};
> get_scalar(..., right_border_values[i])


I'd prefer not to, since we can incorrectly handle stride then. For example stride == -1 (which is not suitable for optimization), dim == 2 and after get_scalar, stride == 1 (which now is suitable for optimization). 
Also we need to know axis first to get the dimension.

But what do you think of something like this:
```
    enum {
        START = 1,
        STOP,
        STRIDE,
        AXIS
    };

    int64_t stride;
    if (!get_scalar(op->get_input_node_shared_ptr(STRIDE), stride) || stride != 1)
        return false;
    if (!get_scalar(op->get_input_node_shared_ptr(AXIS), attrs.axis))
        return false;
    attrs.axis = attrs.axis >= 0 ? attrs.axis : attrs.axis + rank;

    if (input_shape[attrs.axis].is_dynamic())
        return false;
    const auto dimension = input_shape[attrs.axis].get_length();

    for (int i = START; i <= STOP; i++) {
        int64_t value;
        if (!get_scalar(op->get_input_node_shared_ptr(i), value))
            return false;
        value = value >= 0 ? value : value + dimension;
        value = std::min(value, dimension);
        if (i == START)
            attrs.start = value;
        else if (i == STOP)
            attrs.stop = value;
    }
```

Reply:
we can use 0 as a right border for STRIDE for example
value + 0

Reply:
it looks like we need to get axis before the loop in any case
the current impl looks good